Program.cs

```diff
@@ -296,13 +301,22 @@ public void Start()
                     // Get one, show some info
                     _parent.Log("Client accepted:" + clientSocket.RemoteEndPoint.ToString());
                     HTTPProcessor hp = new HTTPProcessor(clientSocket, _parent);
-                    hp.Process();
+                    
+                    Thread t = new Thread(() => foo(hp));
+                    t.Start();
+
                 }
                 catch (Exception ex)
                 {
                     _parent.Log("Server starting error: " + ex.Message + "\n" + ex.StackTrace);
                 }
             }
         }
-    }
+
+        private void foo(HTTPProcessor hp)
```

Comment:
Don't name your method foo (just like many of your friends), no one now what foo means.